### PR TITLE
Improve the error message when source file cannot be read

### PR DESCRIPTION
**Pull request by rylev**
Contribution from @look!

Addresses https://github.com/rust-lang/rust/issues/21329

### CODE REVIEW

**Comment by alexcrichton** [comment]
This has a failing test in the rollup: http://buildbot.rust-lang.org/builders/auto-mac-64-opt/builds/3222/steps/test/logs/stdio

File: src/libsyntax/parse/mod.rs [hunk 1]
```diff
@@ -253,9 +253,10 @@ pub fn file_to_filemap(sess: &ParseSess, path: &Path, spanopt: Option<Span>)
     let bytes = match File::open(path).read_to_end() {
         Ok(bytes) => bytes,
         Err(e) => {
-            err(&format!("couldn't read {:?}: {:?}",
+            let error_msg = e.desc;
+            err(&format!("couldn't read {:?}: {}",
                         path.display(),
-                        e)[]);
+                        error_msg)[]);
```
Comment (alexcrichton) [inline src/libsyntax/parse/mod.rs]: I think this should actually continue to format `e` entirely, it should just use `{}` instead of `{:?}`

**Comment by untitaker** [comment]
Either this got accidentally merged, or https://github.com/rust-lang/rust/issues/21329 can be closed.

**Pull request merged**